docs/TutorialAsync.md

```diff
@@ -46,33 +33,84 @@ export default function request(url) {
 }
 ```
 
-Because we don't want to go to the network in our test, we are going to create
-a manual mock for our `request.js` module in the `__mocks__` folder.
-It could look something like this:
+Next, let's implement a simple async module that fetches user data from an API and
+returns the user name.
 
 ```js
-// __mocks__/request.js
-const users = {
-  4: {name: 'Mark'},
-  5: {name: 'Paul'},
-};
+// src/user.js
+import request from './request';
 
-export default function request(url) {
-  return new Promise((resolve, reject) => {
-    const userID = parseInt(url.substr('/users/'.length), 10);
-    process.nextTick(
-      () => users[userID] ? resolve(users[userID]) : reject({
-        error: 'User with ' + userID + ' not found.',
-      })
-    );
-  });
+export async function getUserName(userID) {
+  const user = await request('/users/' + userID);
+  return user.name;
 }
 ```
 
-Now let's write a test for our async functionality.
+In the above implementation we expect the `request.js` module to return a
+promise. We use the `await` keyword to wait for the promise to resolve before continuing. Because `getUserName` is an async function (defined using the `async` keyword), it will actually return a promise that will resolve with the name when the return statement is reached.
+
+
+### Compile the async module with Babel
+
+To allow modules to use upcoming language features like async/await, we must compile the source into code that our runtimes can handle today. Time to configure Babel:
+
+Start by requiring `babel-cli` from your `package.json` `devDependencies`. Then add a build script that instructs babel to compile the files from your `src` directory into `lib`:
+
+```
+npm install --save-dev babel-cli
+```
+
+```json
+  "devDependencies": {
+    "babel-cli": "*",
+    ...
+  },
+  "scripts": {
+    "build": "babel src --out-dir lib"
+  },
+  "main": "lib/user",
+```
+
+Next we should configure the `.babelrc` inside our project. This configuration transpiles all features of es2015, and compiles async functions into generators:
+
+```json
+{
+  "presets": ["es2015"],
+  "plugins": [
+    "transform-async-to-generator",
+    ["transform-runtime", { "regenerator": true, "polyfill": false }]
```

Comment:
Oops, we've been using the AirBnB style guide on my other project- its got me all confused!